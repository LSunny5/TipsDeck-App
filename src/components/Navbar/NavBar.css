.navBar {
  width: 100%;
  background: lightgrey;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
}

.categoryIcon {
  width: 20%;
  text-decoration: none;
  padding: 3% 0;
  border-right: 1px solid lightslategray;
} 

.backButton{
  padding: 3.5% 0;
  border: none;
  background: none;
  border-right: 1px solid lightslategray;
}

.addIcon {
  border-right: none;
}

.logoImage {
  width: 30px;
  height: 30px;
}

 .buttonName {
  text-decoration: none;
  color: black;
  font-size: 12px;
} 

.dropdown {
  height: 100%;
  width: 100%;
}

.dropButton{
  background-color: lightgrey;
  color: white;
  border: none;
  cursor: pointer;
  padding: 0;
}

.dropdown-content {
  display: none;
  position: absolute;
  background-color: slategray;
  min-width: 90vw;
  right: 0.9rem;
  z-index: 1;
}

/* Show menu on hover */
.dropdown:hover .dropdown-content {
  display: block;
  border-radius: 30px;
}

/* Change color of dropdown links on hover */
.dropdown-content a:hover {
  background-color: darkgray;
}

/* Change color of dropdown links on active and hover */
.dropdown:active, .dropdown:hover, .categoryIcon:active, .categoryIcon:hover,
.dropdown:hover .dropButton {
  background-color: gray;
}

.searchBar {
  margin: 1rem;
}

.searchButton {
  padding: 0.7rem;
}

.searchField {
  font-size: 1.2em;
  text-align: left;
  padding: 3%;
  width: 80%;
}

.searchImage {
  width: 15px;
  height: 15px;
}

.searchResultsBox {
  margin: 10px auto; 
  width: 80%;
  overflow-y: auto;
  max-height: 30rem; 
}

/* for tablets */
@media only screen and (min-width: 600px) {
  .logoImage {
    width: 50px;
    height: 50px;
  }

  .dropdown-content {
    right: 35px;
  }

  .searchField {
    font-size: 1.6em;
    width: 75%;
    border-radius: 10px;
  }

  .searchBar {
    width: 100%;
    margin: auto;
    margin-top: 0.7rem;
  }

  .searchResultsBox {
    display: block;
    background-color: none;
    width: 56%;
    overflow-y: auto;
    max-height: 30rem;
  }

  .cardName {
    margin: 0
  }
}

/* for large screens */
@media only screen and (min-width: 1024px) {
  .logoImage {
    width: 25px;
    height: 25px;
    border: none;
  }

  .categoryIcon {
    border: none;
    width: 9%;
    padding: 2em;
  }

  .navBar {
    display: inline-flex;
    flex-direction: row;
    height: 6rem;
  }

  .backButton {
    position: absolute;
    left: 1rem;
    top: 0;
    margin: 0;
    width: 75px;
    height: 75px;
    top: 1em;
    padding: 0;
  }

  .buttonName {
    font-size: 1em;
    margin-top: 10px;
  }

  .catButton {
    position: absolute;
    right: 15%;
    top: 1.5em;
    padding: 0;
  }

  .randomIcon {
    position: absolute;
    right: 7%;
    top: 1.5em;
    margin: 0;
    padding: 0;
  }

  .addIcon {
    position: absolute;
    right: 0;
    padding: 0;
    margin: 0;
    top: 1.5em;
  }

  .dropButton, .searchWord {
    display: none;
  }

  .dropdown:active, .dropdown:hover, .categoryIcon:active, .categoryIcon:hover,
  .dropdown:hover .dropButton {
    background-color: transparent;
  }

  .addIcon:hover, .randomIcon:hover, .catButton:hover {
    background-color: gray;
    padding: 0;
  }

  .backButton:hover {
    background: gray;
  }

  .searchField{
    font-size: 1em;
    border-radius: 30px;
    margin: 0;
    padding: 1rem;
    width: 95%;
  }

  .dropdown-content {
    display: block;
    background-color: gray;
    min-width: 30rem;
    top: 0.4em;
    right: 35%;
    border-radius: 30px;
  }
  
  .searchResultsBox {
    width: 90%;
    padding: 0 10%;
    overflow-y: auto;
    max-height: 30rem;
    background-color: gray;
  }
}

/* for large screens */
@media only screen and (min-width: 1200px) {
  .dropdown-content {
    min-width: 40rem;
    top: 0.4em;
    right: 35%;
    height: 5rem;
    border-radius: 30px;
  }

  .searchResultsBox {
    padding: 0 20%;
  }
}