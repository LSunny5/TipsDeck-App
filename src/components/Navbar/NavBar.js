import React from 'react';
import './NavBar.css';
import { NavLink } from 'react-router-dom';
import TipsDeckContext from '../../TipsDeckContext';
import Tipcard from '../TipCard/Tipcard'

//Header and navigation bar component and css
class NavBar extends React.Component {
    static contextType = TipsDeckContext;
    constructor() {
        super();
        this.state = {
            search: '',
            filteredTips: [],
        };
    }

    updateSearch(event) {
        this.setState({ search: event.target.value.substr(0, 20) });
    }

    handleClick = () => {
        this.setState({ search: '' });
    }

    render() {
        let term = this.state.search.toLowerCase().trim();
        let filteredTips = [];
        if (term.length !== 0) {
            filteredTips = this.context.tips.filter(tip => {
                return (
                    (tip.tipdescription.toLowerCase().indexOf(term) !== -1)
                    || (tip.tipname.toLowerCase().indexOf(term) !== -1)
                    || (tip.directions.toLowerCase().indexOf(term) !== -1)
                );
            })
        }

        return (
            <nav role="navigation" className="navBar" onClick={this.handleClick}>
                <button
                    className="categoryIcon backButton"
                    onClick={() => this.props.history.goBack()}>
                    <img src="/images/backarrow.png" alt="Go back" className="logoImage back" />
                    <div className="buttonName">Back</div>
                </button>
                <NavLink
                    className="categoryIcon catButton"
                    to={`/Category`}>
                    <img src="/images/CategoriesIcon.png" alt="Categories Icon" className="logoImage" />
                    <div className="buttonName">Categories</div>
                </NavLink>
                <div className="categoryIcon searchContainer">
                    <div className="dropdown" >
                        <button className="dropButton">
                            <img alt="Search Logo" src="/images/MagnifyingIcon.png" className="logoImage" />
                            <div className="buttonName">Search</div>
                        </button>
                        <div className="dropdown-content">
                            <div className="searchResultsBox" >
                                <div className="searchBar" >
                                    <label htmlFor="search">Enter search here...</label>
                                    <input
                                        type="text"
                                        autoComplete="off"
                                        value={this.state.search}
                                        placeholder=" Search for tips here..."
                                        name="search"
                                        className="searchField"
                                        onChange={this.updateSearch.bind(this)}
                                    />
                                </div>
                                <div className = "results" onClick={this.handleClick}>
                                {filteredTips.map((tip) => {
                                    return (
                                        <Tipcard
                                            key={tip.id}
                                            id={tip.id}
                                            name={tip.tipname}
                                            cat={tip.category_id}
                                        />
                                    )
                                })
                                }
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <NavLink
                    className="categoryIcon randomIcon"
                    to={`/Random`}>
                    <img src="/images/dice.png" alt="Random Logo" className="logoImage" />
                    <div className="buttonName">Random</div>
                </NavLink>
                <NavLink
                    className="categoryIcon addIcon"
                    to={`/addTip`}>
                    <img src="/images/PlusSymbol.png" alt="Plus Logo" className="logoImage" />
                    <div className="buttonName">Add Tip</div>
                </NavLink>
            </nav>
        );
    }
}

export default NavBar;