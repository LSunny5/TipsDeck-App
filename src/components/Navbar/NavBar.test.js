import React from 'react';
import ReactDOM from 'react-dom';
import NavBar from './NavBar';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`NavBar component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <NavBar />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the NavBar component', () => {
        const wrapper = shallow(<NavBar />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});

