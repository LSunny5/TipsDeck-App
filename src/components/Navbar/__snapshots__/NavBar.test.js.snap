// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders the NavBar component 1`] = `
<nav
  className="navBar"
  onClick={[Function]}
  role="navigation"
>
  <button
    className="categoryIcon backButton"
    onClick={[Function]}
  >
    <img
      alt="Go back"
      className="logoImage back"
      src="/images/backarrow.png"
    />
    <div
      className="buttonName"
    >
      Back
    </div>
  </button>
  <NavLink
    className="categoryIcon catButton"
    to="/Category"
  >
    <img
      alt="Categories Icon"
      className="logoImage"
      src="/images/CategoriesIcon.png"
    />
    <div
      className="buttonName"
    >
      Categories
    </div>
  </NavLink>
  <div
    className="categoryIcon searchContainer"
  >
    <div
      className="dropdown"
    >
      <button
        className="dropButton"
      >
        <img
          alt="Search Logo"
          className="logoImage"
          src="/images/MagnifyingIcon.png"
        />
        <div
          className="buttonName"
        >
          Search
        </div>
      </button>
      <div
        className="dropdown-content"
      >
        <div
          className="searchResultsBox"
        >
          <div
            className="searchBar"
          >
            <label
              htmlFor="searchText"
            >
              Enter search here...
            </label>
            <input
              autoComplete="off"
              className="searchField"
              id="searchText"
              name="searchText"
              onChange={[Function]}
              placeholder=" Search for tips here..."
              type="text"
              value=""
            />
          </div>
          <div
            className="results"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
  <NavLink
    className="categoryIcon randomIcon"
    to="/Random"
  >
    <img
      alt="Random Logo"
      className="logoImage"
      src="/images/dice.png"
    />
    <div
      className="buttonName"
    >
      Random
    </div>
  </NavLink>
  <NavLink
    className="categoryIcon addIcon"
    to="/addTip"
  >
    <img
      alt="Plus Logo"
      className="logoImage"
      src="/images/PlusSymbol.png"
    />
    <div
      className="buttonName"
    >
      Add Tip
    </div>
  </NavLink>
</nav>
`;
